src/base/bittorrent/torrenthandle.h

```diff
@@ -391,8 +391,8 @@ namespace BitTorrent
         void handleStorageMovedAlert(libtorrent::storage_moved_alert *p);
         void handleStorageMovedFailedAlert(libtorrent::storage_moved_failed_alert *p);
         void handleTrackerReplyAlert(libtorrent::tracker_reply_alert *p);
-        void handleTrackerWarningAlert(libtorrent::tracker_warning_alert *p);
-        void handleTrackerErrorAlert(libtorrent::tracker_error_alert *p);
+        void handleTrackerWarningAlert(const libtorrent::tracker_warning_alert *p);
```

Comment:
Why not `const libtorrent::tracker_warning_alert * const p` ?

Reply:
Or... how about making it `const libtorrent::tracker_warning_alert &p` ?
IMO better use reference than protecting pointer address being changed.

Reply:
Sorry, in written speech the intent was lost.
It was more of a query than a correction proposal.

>Or... how about making it const libtorrent::tracker_warning_alert &p ?

In this case it isn't possible. It comes as a pointer from further up. It would be inefficient to convert it.

Reply:
>In this case it isn't possible. It comes as a pointer from further up. It would be inefficient to convert it.

I think otherwise, just one more change is required, dereference the pointer when calling the function:
`handleTrackerWarningAlert(*static_cast<libt::tracker_warning_alert*>(a));`

Reply:
IMO, I would vote to keep the pointer with a single const and not do the above thing.
Just a personal thing about perceived readability.

Reply:
I would not produce inconsistencies. Either change all similar methods (in separate commit), or don't do it here at all.

Reply:
Added const to other alert handler functions.